src/IdempotencyService.php

```diff
@@ -59,6 +59,13 @@ public function save(RequestInterface $request, ResponseInterface $response)
         $this->idempotentRequestStorage->save($idempotencyKey, $idempotentRequest);
     }
 
+    private function isSuccessfulResponse(ResponseInterface $response) : bool
+    {
+        $statusCode = $response->getStatusCode();
```

Comment:
While the interface is type hinted for integer, there is no proper mechanism in place enforcing the type. Would it be worth casting here?